src/dune/dune_file.ml

```diff
@@ -381,7 +381,26 @@ let enabled_if ~since =
     | None -> Blang.decode
     | Some since -> Dune_lang.Syntax.since Stanza.syntax since >>> Blang.decode
   in
-  field "enabled_if" ~default:Blang.true_ decode
+  let validate enabled_if =
+    Blang.fold_vars enabled_if ~init:(Dune_lang.Decoder.return enabled_if) ~f:(fun var checks ->
+        let loc = String_with_vars.Var.loc var in
+        match
+          ( String_with_vars.Var.payload var
+          , List.assoc_opt (String_with_vars.Var.name var) Lib_config.allowed_in_enabled_if )
+        with
+        | None, Some version ->
+          let desc () = Dune_lang.Decoder.return (loc, "This variable") in
```

Comment:
Nothing is being decoded here, so it feels a bit weird to reuse the parsing function just to raise an error. What about just raising directly with `Dune_lang.Syntax.Error.since`? This is how we validate variables elsewhere I believe.

Reply:
That’s me not properly following the API - I was struggling to see where to get the version of the dune language in force, or is that just `Stanza.syntax`?

If what I’ve written looks weird, assume I was being dopey 🙂

Reply:
Yep, it is. Extensions use different `Dune_lang.Syntax.t` value and the syntax corresponding to the vanilla language is effectively `Stanza.syntax`